crates/subspace-networking/src/node_runner.rs

```diff
@@ -66,7 +63,11 @@ pub struct NodeRunner {
     /// Manages the networking parameters like known peers and addresses
     networking_parameters_registry: Box<dyn NetworkingParametersRegistry>,
     /// Defines set of peers with a permanent connection (and reconnection if necessary).
-    reserved_peers: Vec<Multiaddr>,
+    reserved_peers: HashMap<PeerId, Multiaddr>,
+    /// Incoming swarm connection limit.
+    max_established_incoming_connections: u32,
+    /// Outgoing swarm connection limit.
+    max_established_outgoing_connections: u32,
```

Comment:
With the latest changes these are just connections, not necessarily established

Reply:
Not sure I follow. I check the parameters in the `ConnectionEstablished` event and also check network info counters for established connections. There is a different swarm event for incoming connections. Am I missing something?

Reply:
We do indeed sum those parameters and use `max_established_outgoing_connections` as threshold value on dialing new peers. And that is a side-usage compares to the main limiting use-case when established connections are clear. However, please let me know if yo still think it's confusing and I'll rename both params to max-incoming(outgoing)-connections..